### DISCUSSION TITLE
Framerate of recorded videos

### DISCUSSION THREAD

**Discussion by 765pjj** [roflcoopter/viseron#612] [General]
As a beginner I try to understand the complex system ;-)

What is limiting the framerate of the recorded videos? While the cameras itself (reolink 510a) are recording in 30fps using some object detection, the viseron recording is approx. at 13 fps.

If I switch darknet off in config, the framerate goes up to approx. 18fpx on only motion detection, but not more.

Would more CPU power fix that? Or is there a different limiting factor?
I am running on an Odroid M1 with debian 11. 

Would switching to a different PC help? Would computing on a GPU (like on jetson nano) help on this? Is using the GPU actually working fine in viseron?

**Comment by roflcoopter** [comment]
Viseron saves recordings directly from the camera, so Viserons recordings will always match the source.
Where do you see the numbers 13 and 18?

**Reply by 765pjj** [reply]
When viewing the video it looks like slow motion. Video Lan Client (vlc) shows the frame rate 18 resp.13.

**Reply by roflcoopter** [reply]
can you paste your config please?

**Reply by 765pjj** [reply]
Sure:
[config.yaml.txt](https://github.com/roflcoopter/viseron/files/11589608/config.yaml.txt)

**Comment by 765pjj** [comment]
I edited the fps to 15 later to test. Its the same situation when fps=25 or 30...

**Reply by roflcoopter** [reply]
The FPS that you set in the camera config should match the cameras FPS.
The FPS config parameter only affects motion/object detection tho, recordings are always stored directly from the source.

Could you enable debug logging, restart Viseron and upload it here?

**Reply by 765pjj** [reply]
where do I place this right? Didi it in the nvr section..

logger:
  default_level: info
  cameras:
    camera1: debug

and where do I find the log afterwards?

...now waiting for motion...

**Reply by 765pjj** [reply]
Here is an error log.
From beginning until 6:11 o'clock the recorder is triggerd by motion and object detection,
From 6:11 on the recorder is only triggered by object detection.
FPS is set to 25 at the cameras 1&2 and in the config.
[nohup.out.txt](https://github.com/roflcoopter/viseron/files/11614727/nohup.out.txt)

**Comment by 765pjj** [comment]
[nohup.out.txt](https://github.com/roflcoopter/viseron/files/11610773/nohup.out.txt)

Here is an error log.
From beginning until 6:11 o'clock the recorder is triggerd by motion and object detection,
From 6:11 on the recorder is only triggered by object detection.
FPS is set to 25 at the cameras 1&2 and in the config.

**Comment by roflcoopter** [comment]
I think its the camera acting up.
Do you have any other cameras than Reolinks, and do they work better?

**Reply by 765pjj** [reply]
No, sadly not. Only one Raspberry cam which is considerably slower.
Thanks for looking into this.
 best regards
juliett.papa

**Reply by 765pjj** [reply]
Could you recommend a brand?

**Reply by roflcoopter** [reply]
I am using three different Hikvision domes as well as a doorbell which are working flawlessly.
You can most likely get the Reolinks to work better in Viseron by changing the configuration directly in the cameras and tweaking the FFmpeg parameters in Viseron, but it is very hard for me to recommend anything since i do not have a Reolink.

**Reply by 765pjj** [reply]
Maybe it is  processor power, since all 4 cores are near 100%.  
I can reduce resolution and fps at the cameras but is there an optimal sweetspot for ffmpeg?

**Reply by roflcoopter** [reply]
Yeah maybe it is processor power, never seen that before tho.
There is no particular sweetspot for FFmpeg, but in my experience it is totally unnecessary to have more than around 10 FPS unless you need to catch really fast moving cars for instance.

If i were you i would keep the resolution high but reduce the FPS until it seems to work.
You can also try to use `substream` if your camera has support for that

**Reply by ceewanna** [reply]
For camera, should fps be explicitly specified for both main and substream or should we let ffprobe auto detect? (my config doesn't specify fps.)
For mog2/background extraction, is it correct to say that fps determine how fast motion detection is processed? (my fps is at 2 but for monitoring vehicle passing across gate a value of 5-8 is used.)

I have around 10 cameras, 4 of them being 4MP and the rest 2MP. It seemed the output videos are not progressing smoothly, particularly those cameras with audio. The video player (mpv) always report invalid PTS as the video progresses. The playback fps looks like the factory set value except for those audio-capable cameras will result in slightly lower than their specs (25 by specs turned out to be 19).
